parquet-variant/src/main/java/org/apache/parquet/variant/VariantBuilder.java

```diff
@@ -83,46 +85,28 @@ public Variant build() {
     return new Variant(ByteBuffer.wrap(writeBuffer, 0, writePos), metadataBuffer);
   }
 
-  // Direct access to Metadata.
-  Metadata getMetadata() {
-    return rootBuilder.metadata;
-  }
-
   /**
    * @return the constructed Variant value binary, without metadata.
    */
-  public ByteBuffer valueWithoutMetadata() {
+  public ByteBuffer encodedValue() {
     return ByteBuffer.wrap(writeBuffer, 0, writePos);
   }
 
   /**
    * Directly append a Variant value. Its keys must already be in the metadata
    * dictionary.
    */
-  void appendEncodedValue(ByteBuffer value) {
+  public void appendEncodedValue(ByteBuffer value) {
     onAppend();
     int size = value.remaining();
     checkCapacity(size);
     value.duplicate().get(writeBuffer, writePos, size);
     writePos += size;
   }
 
-  /** Record the current state of the writer. A later call to hasWriteStateChanged() will return true
-   *  if any values have been added since this call.
-   */
-  void markCurrentWriteState() {
-    lastWritePosMark = writePos;
-  }
-
-  /** Returns true if any values have been added since the last call to markCurrentWriteState, or
-   * since the builder was created, if markCurrentWriteState was never called.
-   */
-  boolean hasWriteStateChanged() {
```

Comment:
This is no longer needed by using methods like `appendNullIfEmpty` and `startOrContinueObject`.